database/mockDataGen.js

```diff
@@ -0,0 +1,48 @@
+const faker = require('faker');
+const fs = require('fs');
+const csvWriter = require('csv-write-stream');
+const writer = csvWriter();
+
+// const generateRandomComments = () => {
+//   randomNumber = faker.random.number({
+//     min: 1,
+//     max: 5
+//   });
+//   let comments = [];
+//   for (let i = 0; i < randomNumber; i++) {
+//     comments.push({
+//       username: faker.internet.userName(),
+//       comment: faker.lorem.sentence(),
+//       profileImgUrl: faker.internet.avatar()
+//     });
+//   }
+//   return JSON.stringify(comments);
+// }
+
+
+// let randomComments = generateRandomComments();
+// console.log(randomComments);
+// console.log(randomComments.slice(0, randomComments.length - 2).split(";,"));
+
+const createCSVdata = () => {
+  console.time('timing seed');
+  writer.pipe(fs.createWriteStream('comments.csv'));
+  for (let i = 0; i < 10000000; i++) {
```

Comment:
What happens when you define randomComments inside of the for loop? (i.e. move line 23 inside the for loop) and then add in the array like so ===> {comments: {randomComments}} ? does it crash/not generate?

Reply:
It doesn't crash or not generate, rather lines 23-25 were written for mainly for testing how I would split all the nested data (i.e. multiple comments in one string) client side in future read operations (CRUD). The main factor in whether it crashes or not generates is the number of comments that I'm randomly generating. One to five randomly generate comments for each entry of 10 million entry take 30+ minutes, only about 23 seconds for 1 M though. I don't know the exact time for 10 million with one to five random comments since I stopped it after about 30 minutes. My goal for data generation was under 10 minutes.